pytng/pytng.pyx

```diff
@@ -1037,314 +1136,275 @@ cdef class TNGFileIterator:
+                printf(
+                    "PYTNG ERROR: dtype of array does not match tng dtype\n")
+                return TNG_CRITICAL
+            _float_view = <np.float32_t[:n_atoms, :n_values_per_frame]>(<float*> values)
+            data_loc = np.asarray(_float_view, dtype=np.float32)
 
-        if self.debug:
-            printf("PYTNG INFO: block id %ld \n", self.block_id)
-            printf("PYTNG INFO: data block name %s \n", self.block_name)
-            printf("PYTNG INFO: n_values_per_frame %ld \n", self.n_values_per_frame)
-            printf("PYTNG INFO: n_atoms  %ld \n", self.n_atoms)
-
-        return read_stat
-
-    @cython.boundscheck(True)
-    @cython.wraparound(True)
-    @cython.profile(True)
-    cdef tng_function_status _get_data_next_frame(self, int64_t block_id,
-                                                  int64_t * step,
-                                                  double * frame_time,
-                                                  int64_t * n_values_per_frame,
-                                                  int64_t * n_atoms,
-                                                  double * prec,
-                                                  char * block_name,
-                                                  bint debug) nogil:
+            # for i in range(n_atoms):
+            #     for j in range(n_values_per_frame):
+            #         data[i, j] = ( < float*>values)[i * n_values_per_frame + j]
+
+        elif datatype == TNG_INT_DATA:
+            if dtype != np.int64:
+                printf(
+                    "PYTNG ERROR: dtype of array does not match tng dtype\n")
+                return TNG_CRITICAL
+                _int64_t_view = <np.int64_t[:n_atoms, :n_values_per_frame]>(<int64_t*> values)
+                data_loc = np.asarray(_int64_t_view, dtype=np.int64)
+
+            # for i in range(n_atoms):
+            #     for j in range(n_values_per_frame):
+            #         data[i, j] = (<int64_t*>values)[i * n_values_per_frame + j]
+
+        elif datatype == TNG_DOUBLE_DATA:
+            print("DOUBLE")
+            if dtype != np.float64:
+                printf(
+                    "PYTNG ERROR: dtype of array does not match tng dtype\n")
+                return TNG_CRITICAL
+                _double_view = <np.int64_t[:n_atoms, :n_values_per_frame]>(<double*> values)
+                data_loc = np.asarray(_double_view, dtype=np.float64)
+
+            # for i in range(n_atoms):
+            #     for j in range(n_values_per_frame):
+            #         data[i, j] = (<double*>values)[i * n_values_per_frame + j]
+
+        else:
+            printf("PYTNG WARNING: block datatype not understood")
+            return TNG_CRITICAL
+        
+        #NOTE
+        # couldn't get this to work by assigining the memview to data directly,
+        # seems to require and explicit copy to the destination array
+        np.copyto(data, data_loc, casting="no")
```

Comment:
Correct, perhaps @tylerjereddy would know If this can be done better?

Other options are:
* tying the C array to a `MemoryWrapper` instance and then using PyArray_SetBaseObject, meaning that we do the memory management for the caller (IMO bad idea).
* Iterate over the C array and put it in the NumPy array (FLOP inefficent)
* ??